app/auth/oauth_redis.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cryptography.fernet import Fernet
+import sys
+import time
+
+from flask import current_app
+from oauthlib.oauth2.rfc6749.errors import OAuth2Error
+
+from .oauth_client import RenkuWebApplicationClient
+
+if "pytest" in sys.modules:
+    from fakeredis import FakeStrictRedis as StrictRedis
+else:
+    from redis import StrictRedis
+
+
+class OAuthRedis(StrictRedis):
+    """Just a regular StrictRedis store with extra methods for
+    setting and getting encrypted serializations of oauth client objects."""
+
+    def __init__(self, *args, fernet_key=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fernet = Fernet(fernet_key)
+
+    def set_enc(self, name, value, **kwargs):
```

Comment:
Would be great if we would avoid `**kwargs`. Explicit is better so if those options should be exposed, I suggest we write them explicilty.

Reply:
Agree - I first thought I'd override the `set` and `get` methods, then it would have made sense. But then I decided to keep the original methods around. 
--> Removed those args.